ext/nnstreamer/tensor_decoder/tensordec-protobuf.cc

```diff
@@ -0,0 +1,168 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * GStreamer / NNStreamer tensor_decoder subplugin, "protobuf"
+ * Copyright (C) 2020 Yongjoo Ahn <yongjoo1.ahn@samsung.com>
+ */
+/**
+ * @file        tensordec-protobuf.cc
+ * @date        25 Mar 2020
+ * @brief       NNStreamer tensor-decoder subplugin, "protobuf",
+ *              which converts tensor or tensors to Protocol Buffers.
+ * @see         https://github.com/nnstreamer/nnstreamer
+ * @author      Yongjoo Ahn <yongjoo1.ahn@samsung.com>
+ * @bug         No known bugs except for NYI items
+ *
+ */
+
+#include <glib.h>
+#include <gst/gstinfo.h>
+#include <nnstreamer_plugin_api_decoder.h>
+#include <nnstreamer_plugin_api.h>
+
+#include "nnstreamer.pb.h"    /* Generated by `protoc` */
+
+void init_pb (void) __attribute__ ((constructor));
+void fini_pb (void) __attribute__ ((destructor));
+
+#define DECODER_PB_BIN_CAPS_STR "application/octet-stream"
+
+/**
+ * @brief tensordec-plugin's GstTensorDecoderDef callback
+ */
+static int
+pb_init (void **pdata)
+{
+  /**
+   * Verify that the version of the library we linked is
+   * compatibile with the headers.
+   */
+  GOOGLE_PROTOBUF_VERIFY_VERSION;
+  *pdata = NULL; /* no private data are needed for this sub-plugin */
+  return TRUE;
+}
+
+/** 
+ * @brief tensordec-plugin's GstTensorDecoderDef callback 
+ */
+static void
+pb_exit (void **pdata)
+{
+  google::protobuf::ShutdownProtobufLibrary ();
+  return;
+}
+
+/**
+ * @brief tensordec-plugin's GstTensorDecoderDef callback
+ */
+static int
+pb_setOption (void **pdata, int opNum, const char *param)
+{
+  return TRUE;
+}
+
+/**
+ * @brief tensordec-plugin's GstTensorDecoderDef callback
+ */
+static GstCaps *
+pb_getOutCaps (void **pdata, const GstTensorsConfig * config)
+{
+  return gst_caps_from_string (DECODER_PB_BIN_CAPS_STR);
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static GstFlowReturn
+pb_decode (void **pdata, const GstTensorsConfig * config,
+    const GstTensorMemory * input, GstBuffer * outbuf)
+{
+  GstMapInfo out_info;
+  GstMemory *out_mem;
+  gsize outbuf_size;
+  size_t size;
```

Comment:
FYI, gsize and size_t are same. IMHO, it is better to consistently use one of them for the readability (this is just a personal opinion).

Reply:
I accord the type of two variables. Please check!